### PR TITLE
Handle conn error

### PR DESCRIPTION
**Pull request by giang-nghg**

### CODE REVIEW

**Comment by Zhicheng-Liu** [comment]
@giang-nghg , please fix the build errors.

File: swagger_server/ogm/utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+def with_retry(exception_class, max_retries=3):
+    def outer_wrapper(method):
+        def wrapper(instance, *args, **kwargs):
```
Comment (Zhicheng-Liu) [inline swagger_server/ogm/utils.py:3]:
```suggestion
        def wrapper(instance, *args, **kwargs):
            count = 0
            while count < max_retries:
                try:
                    return method(instance, *args, **kwargs)
                except exception_class:
                    count += 1
            return method(instance, *args, **kwargs)
```

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

File: swagger_server/ogm/utils.py (hunk 1 above)
Comment (Zhicheng-Liu) [reply swagger_server/ogm/utils.py:3]: Sorry I meant to suggest replace the  whole `wrapper` function with the suggested implementation, not just the function declaration line.

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

File: swagger_server/ogm/utils.py (hunk 1 above)
Comment (giang-nghg) [reply swagger_server/ogm/utils.py:3]: I was going for the Decorator Factory pattern: https://www.artima.com/weblogs/viewpost.jsp?thread=240845#decorator-functions-with-decorator-arguments

File: swagger_server/test/ogm/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        with raises(exception_class):
+            with_retry(exception_class, max_retries=max_retries)(instance.method_to_test)(instance)
+
+        assert instance.method_to_test.call_count == max_retries + 1
+
+    inner()
+
+
+def test_retry_decorator_with_no_error(mocker):
```
Comment (Zhicheng-Liu) [inline swagger_server/test/ogm/test_utils.py:34]: Can we add another test case in which an exception that is different from the specified `exception_class` is thrown?

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

**Pull request merged**